File: leetcode-submissions/435_Non-overlapping_Intervals.cpp
class Solution {
public:
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
        int lastpicked =0;
        int count=0;
        for(int i=1;i<intervals.size();i++){
            if(intervals[lastpicked][1]>intervals[i][0]){
                count++;
               
            }
            lastpicked=i;
        }
    return count;
    }
};

File: leetcode-submissions/435_Non-overlapping_Intervals.ts
function eraseOverlapIntervals(intervals: number[][]): number {
    intervals.sort((a,b)=>a[1]-b[1]);
    //after this we need to track of lastend 
    let lastend:number=intervals[0][1];
    let count =0;
    for(let i=1;i<intervals.length;i++){
        if(intervals[i][0]<lastend){
            count++;
        }else{
            lastend=intervals[i][1];

        }
    }
    return count;
};